x-pack/plugins/uptime/public/components/fleet_package/browser/source_field.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiTabbedContent, EuiFormRow, EuiFieldText, EuiSpacer } from '@elastic/eui';
+import { OptionalLabel } from '../optional_label';
+import { CodeEditor } from '../code_editor';
+import { MonacoEditorLangId } from '../types';
+
+enum SourceType {
+  INLINE = 'syntheticsBrowserInlineConfig',
+  ZIP = 'syntheticsBrowserZipURLConfig',
+}
+
+interface SourceConfig {
+  zipUrl: string;
+  folder: string;
+  username: string;
+  password: string;
+  inlineScript: string;
+}
+
+interface Props {
+  onChange: (sourceConfig: SourceConfig) => void;
+  defaultConfig: SourceConfig;
+}
+
+const defaultValues = {
+  zipUrl: '',
+  folder: '',
+  username: '',
+  password: '',
+  inlineScript: '',
+};
+
+export const SourceField = ({ onChange, defaultConfig = defaultValues }: Props) => {
+  const [sourceType, setSourceType] = useState<SourceType>(
+    defaultConfig.inlineScript ? SourceType.INLINE : SourceType.ZIP
+  );
+  const [config, setConfig] = useState<SourceConfig>(defaultConfig);
+
+  useEffect(() => {
+    onChange(config);
+  }, [config, onChange]);
+
+  const tabs = [
+    {
+      id: 'syntheticsBrowserZipURLConfig',
+      name: (
+        <FormattedMessage
```

Comment:
Seems we are repeating this definition in the `content` section below, extract to a common definition above?